src/background_music/Integrator.h

```diff
@@ -0,0 +1,222 @@
+#ifndef AUDACIOUS_PLUGINS_INTEGRATOR_H
+#define AUDACIOUS_PLUGINS_INTEGRATOR_H
+/*
+ * RC-circuit style integrator.
+ * Copyright 2023 Michel Fleur
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions, and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions, and the following disclaimer in the documentation
+ *    provided with the distribution.
+ *
+ * This software is provided "as is" and without any warranty, express or
+ * implied. In no event shall the authors be liable for any damages arising from
+ * the use of this software.
+ */
+#include "utils.h"
+#include <cmath>
+#include <limits>
+#include <stdexcept>
+#include <type_traits>
+
+template<typename S>
+class Integrator
+{
+    static_assert(std::is_floating_point_v<S>);
+
+    S history_multiplier_;
+    S input_multiplier_;
+
+public:
+    /**
+     * Below this fraction of a sample the input multiplier cannot be
+     * distinguished from 1.0 and the history multiplier would become de-normal.
+     */
+    static S minimum_samples()
+    {
+        static S min = -1.0 / (std::log(std::numeric_limits<S>::epsilon()));
```

Comment:
Is there a good reason this code needs to depend on the floating-point epsilon? Can you not just choose safe and reasonable constants for `minimum_samples` and `maximum_samples`?

From experience, code that pushes the limits of the floating-point implementation can be extremely hard to debug. I would like to see a really strong justification for why this is necessary.

Reply:
In this specific case, integration using a floating point number for the integration constants and integrated values becomes unstable for, say, 80 seconds at 96kHz. Inaccuracy is then about <actual seconds>/80 for smaller values, so 10% for 8 seconds. It is, of course, also possible to just use double, in which case the problem goes practically away.